src/query/v2/multiframe.hpp

```diff
@@ -107,22 +107,31 @@ class ValidFramesReader {
     using pointer = value_type *;
     using reference = const Frame &;
 
+    Iterator() {}
     explicit Iterator(FrameWithValidity *ptr) : ptr_(ptr) {}
 
     reference operator*() const { return *ptr_; }
     pointer operator->() { return ptr_; }
 
-    // Prefix increment
     Iterator &operator++() {
       ptr_++;
       return *this;
     }
 
-    friend bool operator==(const Iterator &a, const Iterator &b) { return a.ptr_ == b.ptr_; };
-    friend bool operator!=(const Iterator &a, const Iterator &b) { return a.ptr_ != b.ptr_; };
+    // clang-tidy warning is wrong here, because we & qualify the function, meaning that you can't post increment
+    // temporaries, e.g, (it++)++
+    // NOLINTNEXTLINE (cert-dcl21-cpp)
+    Iterator operator++(int) & {
```

Comment:
I think the `&` qualifier is questionable because of two reasons:
1. In `src/io/network/addrinfo.hppsrc/io/network/addrinfo.hpp` we don't do that, just suppress the warning.
2. It is not done either in [`std` iterators](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:1,endLineNumber:8,positionColumn:1,positionLineNumber:8,selectionStartColumn:1,selectionStartLineNumber:8,startColumn:1,startLineNumber:8),source:'%23include+%3Cvector%3E%0A%0Aint+main()+%7B%0A++++std::vector%3Cint%3E+vec%3B%0A++++auto+it+%3D+(vec.begin()%2B%2B)%2B%2B%3B%0A++++return+*it%3B%0A%7D%0A'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),header:(),k:50,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((g:!((h:compiler,i:(compiler:clang1301,deviceViewOpen:'1',filters:(b:'0',binary:'1',commentOnly:'0',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'-std%3Dc%2B%2B20+-Wall+-Weverything+-Wno-c%2B%2B98-compat+-Wno-c%2B%2B98-compat-pedantic',selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x86-64+clang+13.0.1+(Editor+%231)',t:'0')),header:(),k:50,l:'4',m:50,n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compilerName:'x86-64+clang+13.0.1',editorid:1,fontScale:14,fontUsePx:'0',j:1,wrap:'1'),l:'5',n:'0',o:'Output+of+x86-64+clang+13.0.1+(Compiler+%231)',t:'0')),header:(),l:'4',m:50,n:'0',o:'',s:0,t:'0')),k:50,l:'3',n:'0',o:'',t:'0')),l:'2',n:'0',o:'',t:'0')),version:4), so somebody might be surprised by that.

As this is also not a concern from the correctness point of view, I think the PR can be merged as is and this should be taken care of right after the merge.

Reply:
suppressing the warning can lead to ub. `(it++)++` boom you used a temporary.

Reply:
makes sense, as long as we keep the return type to copy by value